thundra/plugins/invocation/invocation_trace_support.py

```diff
@@ -82,11 +89,25 @@ def get_resources(plugin_context):
             if (not span.get_tag(constants.SpanTags['TOPOLOGY_VERTEX'])
                 or span.span_id == root_span_id):
                 continue
-            rid = resource_id(span)
-            if not rid in resources:
-                resources[rid] = Resource(span)
+
+            resource_names = span.get_tag(constants.SpanTags['RESOURCE_NAMES'])
+            if resource_names:
+                for resource_name in resource_names:
+                    rid = resource_id(span, resource_name)
+                    if rid:
+                        if not rid in resources:
+                            resources[rid] = Resource(span)
+                        else:
+                            resources[rid].merge(span)
+                        resources[rid].name = resource_name
```

Comment:
why we updated resource name each time even it is merged?

Reply:
it is not necessary on merge, will change it